### PR TITLE
EDA-1132: Create Models (US supplier only) - adding taxpayer usa models

### PR DESCRIPTION
**Pull request by sarias-eb**
We are creating the taxpayer form for USA so i'm adding Taxpayer usa model

EDA-1132 https://jira.evbhome.com/browse/EDA-1132

### CODE REVIEW

File: supplier_app/tests/factory_boy.py [hunk 1]
```diff
@@ -13,7 +14,7 @@
     TaxPayer,
     TaxPayerArgentina,
     TaxPayerEBEntity,
-    ContactInformation,
+    ContactInformation, TaxPayerUnitedStates,
```
Comment (ricardocaif-eb) [inline supplier_app/tests/factory_boy.py:17]: NIT: one row below

File: supplier_app/models.py [hunk 2]
```diff
@@ -339,8 +340,54 @@ def get_taxpayer_identifier(self):
...
+        validators=[
+            RegexValidator(
+                r'^[0-9]+$',
+                message=_('Id number must only have numbers'),
+                code='invalid_id'
+            ),
+            MaxLengthValidator(9),
+            MinLengthValidator(9),
+        ]
+    )
```
Comment (ricardocaif-eb) [inline supplier_app/models.py:356]: both EIN and SSN have same validation rules?

File: supplier_app/tests/test_model.py [hunk 3]
```diff
@@ -17,7 +18,7 @@
     TaxPayer,
     TaxpayerComment,
     TaxPayerEBEntity,
-    ContactInformation,
+    ContactInformation, TaxPayerUnitedStates,
```
Comment (ricardocaif-eb) [inline supplier_app/tests/test_model.py:21]: NIT: one row below

File: supplier_app/tests/test_model.py [hunk 4]
```diff
@@ -28,7 +29,7 @@
     TaxPayerFactory,
     TaxPayerArgentinaFactory,
     TaxPayerEBEntityFactory,
-    ContactFactory,
+    ContactFactory, TaxPayerUnitedStatedFactory,
```
Comment (ricardocaif-eb) [inline supplier_app/tests/test_model.py:32]: same here

File: supplier_app/tests/test_view.py [hunk 5]
```diff
@@ -103,7 +103,7 @@
     EBEntityFactory,
     TaxPayerArgentinaFactory,
     TaxPayerEBEntityFactory,
-    ContactFactory,
+    ContactFactory, TaxPayerUnitedStatedFactory,
```
Comment (ricardocaif-eb) [inline supplier_app/tests/test_view.py:106]: NIT: one row below

File: supplier_app/tests/test_view.py [hunk 6]
```diff
@@ -563,7 +563,13 @@ def setUp(self):
...
             company=self.company2,
         )
-
+        self.taxpayer_usa1 = TaxPayerUnitedStatedFactory(
+            workday_id='3',
+            taxpayer_state=STATUS_PENDING,
+            taxpayer_id_number='12345678',
+            company=self.company2,
+            country='USA',
+        )
```
Comment (ricardocaif-eb) [inline supplier_app/tests/test_view.py:572]: are you using it?

File: supplier_app/models.py (hunk 2 above)
Comment (sarias-eb) [reply supplier_app/models.py:356]: Yes

File: supplier_app/tests/test_view.py (hunk 6 above)
Comment (sarias-eb) [reply supplier_app/tests/test_view.py:572]: Yes, It is save for searching data for the table

Comment (ricardocaif-eb) [reply supplier_app/tests/test_view.py:573]: If you delete it, is any going to fail?

**Pull request merged**